eng/Build.props

```diff
@@ -1,9 +1,24 @@
-<Project>
-  <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true' and '$(DotNetBuild)' != 'true'">
+<Project TreatAsLocalProperty="TargetRids">
+  <PropertyGroup>
+    <_AspireOnlyBuild Condition="'$(DotNetBuildFromSource)' != 'true' and '$(DotNetBuild)' != 'true'">true</_AspireOnlyBuild>
```

Comment:
We should clean up the DotnetBuildFromSource and Dotnetbuild from this code. We recently removed aspire from source build so those are not needed any longer.

Reply:
We should just remove this property and instead have a `SkipManagedBuild` property which we check if true and skip if so.